File: programmers 문제 모음/2019 카카오 개발자 겨울 인턴십/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = []
    length = len(board[0])
    answer = 0
    for m in moves: # 인형뽑는 경우
        for y in range(length):
            if board[y][m-1] != 0: # 인형을 뽑은 경우
                catch = board[y][m-1] # catch 변수에 뽑은 인형 저장
                board[y][m-1] = 0
                if len(basket) == 0: # 바구니가 비어있으면 catch 저장
                    basket.append(catch)
                elif basket[-1] == catch: # 바구니의 마지막 값이 catch와 같으면 없애기
                    basket.pop(-1)
                    answer += 2 # 두개의 인형이 없어졌기에 ans 2 증가
                else:
                    basket.append(catch)
                break
    return answer

File: programmers/2019 카카오 개발자 겨울 인턴십/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    let result = [] // 바구니의 역할
    for (let i = 0; i < moves.length; i ++) {
        for (let j = 0; j < board.length; j ++) {
            if (board[j][moves[i] - 1] !== 0) { // 인형을 집어 올리는 경우
                // 이때 바구니 위의 인형과 같은 경우 (인형이 사라지는 경우)
                if (result[result.length - 1] === board[j][moves[i] - 1]) { 
                    result.pop()
                    answer += 2
                }
                // 바구니 맨 위의 인형과 다른 경우
                else {
                    result.push(board[j][moves[i] - 1])
                }
                
                board[j][moves[i] - 1] = 0
                break
            }
        }
    }
    return answer;
}

// console.log(solution([[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]], [1,5,3,5,1,2,1,4]))